disk.c

```diff
@@ -1213,6 +1244,123 @@ void wd17xx_write( struct machine *oric, struct wd17xx *wd, unsigned short addr,
             wd->delayeddrq = 32;
           }
           break;
+
+        case COP_WRITE_TRACK:
+          switch(data)
+          {
+            // All bytes > 0xF4 are control bytes
+            case 0xf5:
+              // MFM: Initialize CRC generator
+              // FM : Not allowed
+#if GENERAL_DISK_DEBUG
+              dbg_printf( "\tCOP_WRITE_TRACK: %02X -> SYNC (clear crc) -> 0xA1", wd->r_data);
+#endif
+              wd->crc = 0x968b;
+              wd->r_data = 0xa1;
+              wd->crc = calc_crc( wd->crc, wd->r_data );
+              // wd->crc = 0xcdb4;
+              break;
+
+            case 0xf6:
+              // FM : Not allowed
+#if GENERAL_DISK_DEBUG
+              dbg_printf( "\tCOP_WRITE_TRACK: %02X -> SYNC -> 0xC2", wd->r_data);
+#endif
+              wd->r_data = 0xc2;
+              wd->crc = calc_crc( wd->crc, wd->r_data );
+              break;
+
+            case 0xf7:
+#if GENERAL_DISK_DEBUG
+              dbg_printf( "\tCOP_WRITE_TRACK: %02X -> CRC -> %04X", wd->r_data, wd->crc);
+#endif
+              wd->disk[wd->c_drive]->rawimage[(wd->c_side*wd->disk[wd->c_drive]->numtracks + wd->c_track)*6400+256 + wd->curroffs] = wd->crc >> 8;
+              wd->curroffs++;
+              wd->r_data = wd->crc & 0xff;
+              break;
+
+#if GENERAL_DISK_DEBUG
+            // MFM: 0xf8 -> 0xff: write control byte
+            // FM : 0xf8 -> 0xfe: Initialize CRC generator
+            case 0xf8:
+              dbg_printf( "\tCOP_WRITE_TRACK: %02X -> Data Address Mark (deleted)", wd->r_data);
+              wd->crc = calc_crc( wd->crc, wd->r_data );
```

Comment:
I think this (and correspondingly for F9, FA, FB and FE) will act incorrectly when formatting as single density: in FM mode those values should reset the CRC generator — a genuine reset to 0xffff, not a loading of the special "assume it'll be three A1s" CRC value that's used in double density mode.

Reply:
Yes, you're right and it's in the comment line:
```
// FM : 0xf8 -> 0xfe: Initialize CRC generator
```
But, AFAIK, there is no way to write in single density mode with the Microdisc or the Jasmin interfaces because the -DDEN pin of the FDC is hard wired to GND, so the FDC is forced to Double Density mode.